crates/relayer-types/src/core/ics24_host/identifier.rs

```diff
@@ -442,3 +454,26 @@ impl Display for PortChannelId {
         write!(f, "{}/{}", self.port_id, self.channel_id)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_chain_id_name() {
+        let chain_id = ChainId::from_str("ibc-5").unwrap();
+        assert_eq!(chain_id.name(), "ibc".to_owned());
+    }
+
+    #[test]
+    fn test_chain_id_name_with_test() {
+        let chain_id = ChainId::from_str("ibc-test-5").unwrap();
+        assert_eq!(chain_id.name(), "ibc-test".to_owned());
+    }
+
+    #[test]
+    fn test_chain_id_name_no_dash() {
+        let chain_id = ChainId::from_str("ibc5").unwrap();
+        assert_eq!(chain_id.name(), "ibc5".to_owned());
+    }
```

Comment:
Let's maybe add a few more tests:

```rust
#[test]
fn standard_chain_names() {
    let test_cases = [
        ("ibc-0", "ibc"),
        ("osmosis-1", "osmosis"),
        ("cosmoshub-4", "cosmoshub"),
        ("juno-mainnet-1", "juno-mainnet"),
        ("akash-testnet-2", "akash-testnet"),
        ("stargaze-123", "stargaze"),
        ("crypto-org-chain-0", "crypto-org-chain"),
        ("mars-hub-1", "mars-hub"),
        ("evmos-9000-1", "evmos-9000"),
    ];

    for (input, expected) in test_cases {
        let chain_id = ChainId::from_str(input).unwrap();
        assert_eq!(chain_id.name(), expected);
    }
}

#[test]
fn missing_or_invalid_revision_numbers() {
    let test_cases = [
        ("osmosis", "osmosis"),
        ("ibc-test", "ibc-test"),
        ("cosmos-hub", "cosmos-hub"),
        ("juno-", "juno-"),
        ("akash-testnet-x", "akash-testnet-x"),
        ("stargaze-abc", "stargaze-abc"),
        ("chain-123-xyz", "chain-123-xyz"),
        ("-1", "-1"),
        ("chain--1", "chain--1"),
    ];

    for (input, expected) in test_cases {
        let chain_id = ChainId::from_str(input).unwrap();
        assert_eq!(chain_id.name(), expected);
    }
}

#[test]
fn edge_cases() {
    let test_cases = [
        ("", ""),
        ("-", "-"),
        ("chain-", "chain-"),
        ("chain-9999999999999999999999", "chain-9999999999999999999999"),  // number too large for u64
    ];

    for (input, expected) in test_cases {
        let chain_id = ChainId::from_str(input).unwrap();
        assert_eq!(chain_id.name(), expected);
    }
}
```

Not sure exactly if the expected values are correct for the weird ones. Feel free to adapt the tests